library/oneview_id_pools_ipv4_range.py

```diff
@@ -91,43 +91,72 @@ def __init__(self):
 
         super(IdPoolsIpv4RangeModule, self).__init__(additional_arg_spec=additional_arg_spec,
                                                      validate_etag_support=True)
-
+        self.connection_templates = self.oneview_client.connection_templates
         self.resource_client = self.oneview_client.id_pools_ipv4_ranges
 
     def execute_module(self):
         resource = None
+        # If Range URI is provided then it sets the resource client
         if self.data.get('uri'):
-            resource = self.resource_client.get(self.data.get('uri'))
+            resource = self.resource_client.get_by_uri(self.data.get('uri'))
```

Comment:
why don't you use "set_resource_object" method in __init__ function, so that it will do these checks (get_by_name and get_by_uri)..?
Also instead of using resource variable, assign the response of get_by_uri to class variable(self.current_resource) so that it will have its scope throughout the class.

Reply:
I am afraid we cant use it, because in case for get_by_name, range URIs are specific with subnet URIs, so i had to explicitly call them differently.